mapfile.c

```diff
@@ -1351,6 +1351,44 @@ int msLoadProjectionStringEPSG(projectionObj *p, const char *value)
   return msLoadProjectionString(p, value);
 }
 
+int msLoadProjectionCodeString(projectionObj* p, const char* value) {
+
+  int num_params = 0;
+
+  // exit if init= is already at the start of the string e.g. from msOGRSpatialRef2ProjectionObj
+  if (strncasecmp(value, "init=", 5) == 0) {
+      return -1;
+  }
+
+  if (!strchr(value, ':')) {
+      return -1;
+  }
+
+  char** papszList = NULL;
+  papszList = msStringSplit(value, ':', &(num_params));
+
+  if (num_params != 2) {
+      return -1;
+  }
+
+  size_t buffer_size = 0;
+  char* init_string = NULL;
+  buffer_size = 5 + strlen(value) + 1;
+  init_string = (char*)msSmallMalloc(buffer_size);
+
+  /* translate into PROJ format. */
+  msStringToLower(papszList[0]);
+  snprintf(init_string, buffer_size, "init=%s:%d", papszList[0], atoi(papszList[1]));
+  
+  p->args = (char**)msSmallMalloc(sizeof(char*) * 2);
```

Comment:
I know you took inspiration from code above, but it doesn't look right to me.
It should be either:
- null terminate the array:
```suggestion
  p->args = (char**)msSmallCalloc(sizeof(char*), 2);
```
- or
```suggestion
  p->args = (char**)msSmallMalloc(sizeof(char*));
```
I would say the later is probably the more logical given that (I believe that) args[] is not assumed to be NULL terminated.